deps/src/main/java/com/microsoft/azure/sdk/iot/deps/auth/IotHubCertificateManager.java

```diff
@@ -105,83 +112,49 @@
+        return this.certificates;
     }
 
     /**
      * Setter for a valid certificate path provided by user
      * @param certPath The valid path where a certificate could be loaded from.
      * @throws IOException If the certPath was null or could not be read or was empty.
      */
-    void setValidCertPath(String certPath) throws IOException
+    void setCertificatesPath(String certPath) throws IOException, CertificateException
     {
-        //Codes_SRS_IOTHUBCERTIFICATEMANAGER_25_006: [*If a user attempted to set the certificate and for somereason could not succeed then this method shall not use default certificate by setting valid certificate as null.**]**
-        this.validCert = null;
-
-        if (certPath == null)
+        //Codes_SRS_IOTHUBCERTIFICATEMANAGER_34_007: [**This method shall throw IllegalArgumentException if the cert path is null or empty.**]**
+        if (certPath == null || certPath.isEmpty())
         {
-            //Codes_SRS_IOTHUBCERTIFICATEMANAGER_25_002: [**This method shall throw IllegalArgumentException if parameter is null.**]**
-            throw new IllegalArgumentException("Cert path cannot be null");
+            throw new IllegalArgumentException("Cert path cannot be null or empty");
         }
 
-        //Codes_SRS_IOTHUBCERTIFICATEMANAGER_25_003: [**This method shall attempt to read the contents of the certificate file from the path provided and save it as valid certificate.**]**
-        //Codes_SRS_IOTHUBCERTIFICATEMANAGER_25_005: [**This method shall throw FileNotFoundException if it could not be found or does not exist.**]**
-        try (BufferedReader certReader = new BufferedReader(new FileReader(certPath)))
+        //Codes_SRS_IOTHUBCERTIFICATEMANAGER_34_004: [**This function shall read the certificates from the provided
+        // path and save them into a certificate collection.**]**
+        try (FileInputStream fis = new FileInputStream(certPath))
         {
-            long readLineCount = 0;
-            String line;
-            StringBuilder cert = new StringBuilder();
-            while ((line = certReader.readLine()) != null)
-            {
-                if (readLineCount > MAX_CERTIFICATE_LINES)
-                {
-                    throw new IllegalArgumentException("The provided certificate is too long");
-                }
-
-                cert.append(line);
-                cert.append("\r\n");
-                readLineCount++;
-            }
-
-            if (cert.length() == 0)
-            {
-                //Codes_SRS_IOTHUBCERTIFICATEMANAGER_25_004: [**This method shall throw IllegalArgumentException if certificate contents were empty.**]**
-                throw new IllegalArgumentException("Cert cannot be Empty");
-            }
-
-            this.validCert = cert.toString();
+            certificates = certificateFactory.generateCertificates(fis);
```

Comment:
Check `certificates` to not be empty will still be required.

Reply:
sure thing